src/NServiceBus.Core/Routing/AssemblyRouteSource.cs

```diff
@@ -20,7 +20,7 @@ public AssemblyRouteSource(Assembly messageAssembly, UnicastRoute route)
         public IEnumerable<RouteTableEntry> GenerateRoutes(Conventions conventions)
         {
             var routes = messageAssembly.GetTypes()
-                .Where(t => conventions.IsMessageType(t))
+                .Where(conventions.IsMessageType)
```

Comment:
@SimonCropp I think if the arguments all speak for using a delegate as it would cause less allocations and the code to perform better and no guideline states that you should use method groups where possible, then it makes sense to do that change. Also I'm pretty sure I'll find quite a few delegates instead of method groups in the code base, so there is no consistency which you are referring to. If you argue that we then should move over the existing method groups to delegates, I'd most likely agree with you but this seems to be out of scope of this PR and we would happily accept a PR containing that change (or an issue to track this work item).

If you don't mind to use delegates instead of the method group, I'm happy to provide that change to this PR so we can go ahead and merge it. Thoughts?